content/developer/reference/frontend/odoo_editor.rst

```diff
@@ -0,0 +1,223 @@
+
+Powerbox
+========
+
+The Powerbox is a piece of user interface that contains *commands* organized
+into *categories*. The commands can be filtered when the user inputs text, and
+navigated with the arrow keys.
+
+.. image:: editor/powerbox.png
+
+Modifying the Powerbox
+----------------------
+
+Only one Powerbox should be instantiated at the time, and that job is done by
+the editor itself. Its Powerbox instance can be found in its `powerbox` instance
+variable.
+To change the Powerbox's contents and options, you will need to change the
+options passed to the editor before it gets instantiated.
+
+.. important::
+   Never instantiate the Powerbox yourself, always use the current editor's own
+   instance instead.
+
+Example: adding a category and a command
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Say we want to add a new command `Document` to the Powerbox, just for the
+`mass_mailing` module. We want to add it to a new category called
+`Documentation` and we want it all the way at the top of the Powerbox.
+
+`mass_mailing` `extends
+<{GITHUB_PATH}/addons/mass_mailing/static/src/js/wysiwyg.js>`_
+`web_editor`'s `Wysiwyg class
+<{GITHUB_PATH}/addons/web_editor/static/src/js/wysiwyg/wysiwyg.js>`_, which
+instantiates the editor in its `start` method. Before doing so, it
+calls its own `_getPowerboxOptions` method, which can conveniently be overridden
+to add our new commands.
+
+Since `mass_mailing` already overrides `_getPowerboxOptions`, let's just add our
+new command to it:
+
+.. code-block:: javascript
+
+   _getPowerboxOptions: function () {
+       const options = this._super();
+       // (existing code before the return statement)
+       options.categories.push({
+           name: _t('Documentation'),
+           priority: 300,
+       });
+       options.commands.push({
+           name: _t('Document'),
+           category: _t('Documentation'),
+           description: _t("Add this text to your mailing's documentation"),
+           fontawesome: 'fa-book',
+           priority: 1, // This is the only command in its category anyway.
+       });
+       return options;
+   }
+
+The :ref:`reference <reference>` for the definition of :ref:`commands <command>`
+and :ref:`categories <category>` can be found further on this page.
+
+.. note::
+   Think of wrapping the names and descriptions of your commands and categories
+   in the `_t` function for translation purposes.
+
+.. note::
+   To avoid out-of-control escalations, don't use random numbers for your
+   priorities: look at what other priorities already exist and choose your value
+   accordingly (like you would do for a z-index).
+
+Opening a custom Powerbox
+-------------------------
+
+Say we want to open the Powerbox to offer a couple of options, no more, and we
+don't need all the clutter of the other commands. This is possible by manually
+calling the `open` method of the Powerbox and passing it the commands and
+categories we want.
```

Comment:
Try to formulate this so that it sounds less like an exercise and more like a fact: "It is possible to open the Powerbox with a custom set of categories and commands...blabla".